### ISSUE TITLE
SQL tests not being executed

### ISSUE THREAD

**Issue by datapythonista** [ibis-project/ibis#2477]
There is something I don't understand going on with the tests. Checking the files in `ibis/tests`, this is what we've got:
```
(ibis-dev) [mgarcia@xps ibis]$ find ibis/tests/ -name "*.py"
ibis/tests/test_filesystems.py
ibis/tests/test_version.py
ibis/tests/all/test_string.py
ibis/tests/all/test_join.py
ibis/tests/all/test_json.py
ibis/tests/all/test_param.py
ibis/tests/all/test_numeric.py
ibis/tests/all/conftest.py
ibis/tests/all/test_nettypes.py
ibis/tests/all/test_generic.py
ibis/tests/all/test_array.py
ibis/tests/all/test_window.py
ibis/tests/all/test_aggregation.py
ibis/tests/all/test_temporal.py
ibis/tests/all/test_vectorized_udf.py
ibis/tests/all/__init__.py
ibis/tests/all/test_column.py
ibis/tests/all/test_union.py
ibis/tests/all/test_client.py
ibis/tests/all/test_geospatial.py
ibis/tests/backends.py
ibis/tests/util.py
ibis/tests/sql/test_sqlalchemy.py
ibis/tests/sql/__init__.py
ibis/tests/sql/test_compiler.py
ibis/tests/expr/test_string.py
ibis/tests/expr/test_datatypes.py
ibis/tests/expr/test_signature.py
ibis/tests/expr/test_timestamp.py
ibis/tests/expr/test_table.py
ibis/tests/expr/test_analysis.py
ibis/tests/expr/test_interactive.py
ibis/tests/expr/test_lineage.py
ibis/tests/expr/mocks.py
ibis/tests/expr/test_format.py
ibis/tests/expr/test_case.py
ibis/tests/expr/test_decimal.py
ibis/tests/expr/test_schema.py
ibis/tests/expr/conftest.py
ibis/tests/expr/test_visualize.py
ibis/tests/expr/test_array.py
ibis/tests/expr/test_temporal.py
ibis/tests/expr/test_rules.py
ibis/tests/expr/test_literal.py
ibis/tests/expr/test_sql_builtins.py
ibis/tests/expr/test_analytics.py
ibis/tests/expr/test_window_functions.py
ibis/tests/expr/__init__.py
ibis/tests/expr/test_pipe.py
ibis/tests/expr/test_operations.py
ibis/tests/expr/test_geospatial.py
ibis/tests/expr/test_value_exprs.py
ibis/tests/__init__.py
```

So, when we execute pytest for that dir, the files in `all`, `sql` and `expr` besides the files in the root should be executed. But for some reason the tests in `sql` are not being executed, see:

```
(ibis-dev) [mgarcia@xps ibis]$ PYTEST_BACKENDS="sqlite" python -m pytest ibis/tests/
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.8.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/mgarcia/src/ibis, configfile: setup.cfg
plugins: mock-3.3.1, cov-2.10.1
collected 3074 items / 4 skipped / 3070 selected                                                                                                                                                                                             

ibis/tests/test_filesystems.py .EEEEEEEEEEEEEEEEEEEEEEEE                                                                                                                                                                               [  0%]
ibis/tests/test_version.py ..                                                                                                                                                                                                          [  0%]
ibis/tests/all/test_aggregation.py .x....x..............x....xxxxx...........x....xxxxx..s                                                                                                                                             [  2%]
ibis/tests/all/test_array.py xx                                                                                                                                                                                                        [  2%]
ibis/tests/all/test_client.py ....ss.x.xs                                                                                                                                                                                              [  3%]
ibis/tests/all/test_column.py ...s..                                                                                                                                                                                                   [  3%]
ibis/tests/all/test_generic.py ....................                                                                                                                                                                                    [  3%]
ibis/tests/all/test_join.py ssssss                                                                                                                                                                                                     [  4%]
ibis/tests/all/test_json.py ss                                                                                                                                                                                                         [  4%]
ibis/tests/all/test_nettypes.py ss                                                                                                                                                                                                     [  4%]
ibis/tests/all/test_numeric.py xxxxxxxxxxxx.................xx..........................ssssssssssssss                                                                                                                                 [  6%]
ibis/tests/all/test_param.py ....xx.xxx                                                                                                                                                                                                [  6%]
ibis/tests/all/test_string.py .....x...xxx........xx.................xxs                                                                                                                                                               [  8%]
ibis/tests/all/test_temporal.py xxx......x...x...xxxxxxxxxxxxxxxxxxxxxxxxxx.....xxxxxxxxx.xxxs............                                                                                                                             [ 10%]
ibis/tests/all/test_union.py ss                                                                                                                                                                                                        [ 10%]
ibis/tests/all/test_vectorized_udf.py ssssssssss                                                                                                                                                                                       [ 11%]
ibis/tests/all/test_window.py ....xx........x.x......x....x.x.x..x.x..x                                                                                                                                                                [ 12%]
ibis/tests/expr/test_analysis.py .......xx...                                                                                                                                                                                          [ 12%]
ibis/tests/expr/test_analytics.py ......                                                                                                                                                                                               [ 12%]
ibis/tests/expr/test_array.py ....                                                                                                                                                                                                     [ 13%]
ibis/tests/expr/test_case.py .x..x..xx.                                                                                                                                                                                                [ 13%]
ibis/tests/expr/test_datatypes.py .............................................................................................................................................................xxxxx.........................          [ 19%]
ibis/tests/expr/test_decimal.py .................                                                                                                                                                                                      [ 20%]
ibis/tests/expr/test_format.py ................                                                                                                                                                                                        [ 20%]
ibis/tests/expr/test_interactive.py .X........                                                                                                                                                                                         [ 20%]
ibis/tests/expr/test_lineage.py .......                                                                                                                                                                                                [ 21%]
ibis/tests/expr/test_literal.py .....                                                                                                                                                                                                  [ 21%]
ibis/tests/expr/test_operations.py .......                                                                                                                                                                                             [ 21%]
ibis/tests/expr/test_pipe.py ....                                                                                                                                                                                                      [ 21%]
ibis/tests/expr/test_rules.py .....................................................x.............................                                                                                                                      [ 24%]
ibis/tests/expr/test_schema.py ....                                                                                                                                                                                                    [ 24%]
ibis/tests/expr/test_signature.py ....................................                                                                                                                                                                 [ 25%]
ibis/tests/expr/test_sql_builtins.py ...................xx                                                                                                                                                                             [ 26%]
ibis/tests/expr/test_string.py .........                                                                                                                                                                                               [ 26%]
ibis/tests/expr/test_table.py ............x.....x....................................x.x...x...................................x.......xx.....x...............                                                                         [ 30%]
ibis/tests/expr/test_temporal.py ...................................................................................xx................................................................................................................ [ 37%]
...................................................................................................................................................................................................................................... [ 44%]
...................................................................................................................................................................................................................................... [ 52%]
...................................................................................................................................................................................................................................... [ 59%]
...................................................................................................................................................................................................................................... [ 67%]
...................................................................................................................................................................................................................................... [ 74%]
....                                                                                                                                                                                                                                   [ 74%]
ibis/tests/expr/test_timestamp.py .............x......................                                                                                                                                                                 [ 75%]
ibis/tests/expr/test_value_exprs.py .x..............................................................xxx.xx..x.....x........x..........................................xxxx..............x........x.................................... [ 82%]
...................................................................................................................................................................................................................................... [ 89%]
...................................................................xxxxx.............................................................................................................................................................. [ 97%]
.......................x......................xxxxxx...                                                                                                                                                                                [ 98%]
ibis/tests/expr/test_visualize.py ................                                                                                                                                                                                     [ 99%]
ibis/tests/expr/test_window_functions.py ...............                                                                                                                                                                               [100%]

=================================================================================================================== ERRORS ===================================================================================================================
```

**Comment by renato2099** [comment]
the problem seems to be that `pytest.importorskip('impala.dbapi')` fails and all the tests are skipped.
the solution is also quite simple, just do `pip install impyla` and then tests are no longer skipped.
regarding CI, we might want to add it to the `environment.yaml` file. I put a PR on this here https://github.com/ibis-project/ibis/pull/2510
let me know what you think @datapythonista

**Comment by datapythonista** [comment]
Thanks @renato2099. I expected pytest to run the files and show them there in that log, and then show them as skipped. Which I think it's not true when the skip is at the beginning of the file.

Impala shouldn't be required there, but it's being removed in #2501 after some refactoring. This is fixed there, just checked its CI.

Thanks for researching this!

**Comment by renato2099** [comment]
sounds good @datapythonista!
yeah I wasn't sure what you meant, and I thought that we couldn't simply run the sql tests, and in my case it was because I was missing `impyla` package, once I installed, those tests were no longer skipped. But if that is fixed by the refactorings in #2501 that is even better!

**Comment by cpcloud** [comment]
This doesn't appear to be a problem any more.
